the-collab-lab/tcl-19-smart-shopping-list
# Ju cu join existing list

## Description

The PR updates the existing component `Home.js`: A form is created where an existing list can be joined by entering a shared token, which will be saved to localStorage after hitting submit. The existing list then will be shown.

If no token is entered, a message appears which says, that you should enter a token.
If the entered token is wrong, a message appears which says, that the token does not exist and you should either try again or create a new list.  

**Bugfixes**
Besides, the bug about not updating the list automatically was fixed by adding react-firebase-hooks again.

## Related Issue
Closes #5 

## Acceptance Criteria

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|   ✓  | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates
A simple style was added to the homepage with the new form for shared tokens.
### Before
[![Homepage_alt](https://docs.google.com/uc?export=download&id=1yht8h5D6qBTWuctCF1hotJ6qZOAChG3Q)

### After
[![Homepage](https://docs.google.com/uc?export=download&id=1HaGcq-zuG9AWkwm3po_-T8W8YcoeWZvr)


## Testing Steps / QA Criteria
- From your terminal, pull down this branch with `git pull origin ju-cu-join-existing-list`
- check that branch out with git checkout `ju-cu-join-existing-list`
- Then `npm install` is case new dependencies were added and `npm start` to launch the app.
- Try to hit `submit` without entering a token to see the alert message
- Try to enter the wrong token to see the alert message
- Enter the shared token **hinge grater hhhh** to get access to the existing list. This token now is saved in `localStorage`
